pkg/executor/aggregate/agg_hash_final_worker.go

```diff
@@ -151,18 +165,70 @@ func (w *HashAggFinalWorker) loadFinalResult(sctx sessionctx.Context) {
 		}
 
 		if result.IsFull() {
+			if w.checkFinishChClosed() {
```

Comment:
Is this if check needed.
We already check finishCh in line 173

Reply:
> Is this if check needed. We already check finishCh in line 173

I will delete it.